### ISSUE TITLE
[BUG] Tuya TY0201 Temperature/Humidity sensor does not work

### ISSUE THREAD

**Issue by vJan00** [zigpy/zha-device-handlers#3230]
### Bug description

I have tested the Quirk which was merged by #3201 and it does not work with my TY0201 _TZ3000_bjawzodf!
It does not show humidity, nor the temperature, which it showed without the Quirk.
Based on other Quirks I have developed my own Quirk for TY0201 _TZ3000_bjawzodf, which works without problems.
This should be integrated as V2.

### Steps to reproduce

Try to pair the TY0201 _TZ3000_bjawzodf and read temperature or humidity sensor (both doesn't show up).

### Expected behavior

Pairing the TY0201 _TZ3000_bjawzodf which results in temperature and humidity sensor showing up.

**Comment by vJan00** [comment]
Opened #3231 to resolve this issue.

**Comment by vJan00** [comment]
@philippedeswert would it be possible you can test the updated ty0201.py too? :)

**Comment by philippedeswert** [comment]
> @philippedeswert would it be possible you can test the updated ty0201.py too? :)

Sure, I was planning to do the v2 rewrite myself this weekend as I was getting an extra HA setup so I did not need to mess with my "production" setup to test things.

**Comment by Mattie112** [comment]
Do you happen to have any progress on this? I bought a couple but (in HomeAssisant) only the temperature shows up, not the humidity.

**Comment by philippedeswert** [comment]
I haven't had time yet unfortunately. I managed to find a little bit of time to test things, but something is indeed still off as I added a new one, on a clean home assistant instance and I had to restart zha to have it recognize everything.

**Comment by Mattie112** [comment]
Yeah, I just wanted to add: added 2more but those don't even have the temperature shown. Strange... I will see if I can dive into it but not sure where to start.

**Comment by philippedeswert** [comment]
Try reconfigure in the device menu and then restart zha...

**Comment by Mattie112** [comment]
Ah yes indeed, now it shows up. We'll great thanks! Let's see if it continues to work :)

**Comment by phhusson** [comment]
I don't really understand. I have the same issue. I've tried reconfiguring it three times with always the same result.

But I can read the TuyaAirQualityTemperature and TuyaAirQualityHumidity clusters just fine manually?

![2024-10-31_00-03_2](https://github.com/user-attachments/assets/e3209737-3faa-4d09-a917-d557ffa49fb9)
![2024-10-31_00-03_1](https://github.com/user-attachments/assets/4eadcaa2-c8d5-42c2-9cde-bc471669c38b)
![2024-10-31_00-03](https://github.com/user-attachments/assets/14ba5417-2871-4dc8-a1c7-bf9af460b2e1)

Can someone point me to what am I doing wrong?


Edit: @vJan00 's change seem to fix this issue for me

**Comment by Migsi** [comment]
I can also confirm this fix works. @vJan00 please look forward to get your PR rebased and merged ;)
